tests/baselines/reference/propertyAccessChain.types

```diff
@@ -89,3 +89,12 @@ o1?.b ? 1 : 0;
 >1 : 1
 >0 : 0
 
+// GH#36031
+o2?.b!.c
+>o2?.b!.c : string

```

Comment:
This should be `string | undefined`, shouldn't it? That means type checking for non-null assertions also needs to be changed

Reply:
This fix is just intended to address the js emit. The resulting types for optional chaining is a different set of issues.